pennylane/operation.py

```diff
@@ -656,10 +674,13 @@ def eigvals(self):
 
         try:
             return self.compute_eigvals(*self.parameters, **self.hyperparameters)
-        except NotImplementedError:
+        except EigvalsUndefinedError:
             # By default, compute the eigenvalues from the matrix representation.
             # This will raise a NotImplementedError if the matrix is undefined.
-            return np.linalg.eigvals(self.matrix())
+            try:
+                return np.linalg.eigvals(self.matrix())
+            except MatrixUndefinedError as e:
+                raise EigvalsUndefinedError from e
```

Comment:
Do you think it's bad practise to leave this as-is? I suppose someone could be confused by `MatrixUndefined` being raised from a call to `eigvals()`?

Or maybe it makes sense, since it makes the logical flow clear?

Reply:
In the latter case, a developer could just do `except OperatorPropertyUndefined:`

Reply:
I think this is cleaner, if PL core wants to except an eigval error!